packages/angular_devkit/build_angular/test/browser/common-js-warning_spec_large.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Architect } from '@angular-devkit/architect';
+import { logging } from '@angular-devkit/core';
+import { createArchitect, host } from '../utils';
+
+describe('Browser Builder commonjs warning', () => {
+  const targetSpec = { project: 'app', target: 'build' };
+
+  let architect: Architect;
+  let logger: logging.Logger;
+  let logs: string[];
+
+  beforeEach(async () => {
+    await host.initialize().toPromise();
+    architect = (await createArchitect(host.root())).architect;
+
+    // Add a Common JS dependency
+    host.appendToFile('src/app/app.component.ts', `import 'bootstrap';`);
+
+    // Create logger
+    logger = new logging.Logger('');
+    logs = [];
+    logger.subscribe(e => logs.push(e.message));
+  });
+
+  afterEach(async () => host.restore().toPromise());
+
+  it('should show warning when depending on a Common JS bundle', async () => {
+    const run = await architect.scheduleTarget(targetSpec, undefined, { logger });
+    const output = await run.result;
+    expect(output.success).toBe(true);
+    expect(logs.join()).toMatch(/WARNING in.+app\.component\.ts depends on bootstrap\. CommonJS or AMD dependencies/);
+    await run.stop();
+  });
+
+  it('should not show warning when depending on a Common JS bundle which is allowed', async () => {
+    const overrides = {
+      allowedCommonJsDepedencies: [
```

Comment:
@filipesilva / @clydin should allowing a CommonJS dep allow also it's CommonJS transitive dependency automatically?

Reply:
Hmm that's a tricky one. Transitive dependencies in general really. 

Realistically speaking many project, especially larger ones, will have dozens of transitive commonjs dependencies. So they get these changes and see a screenful of names they have to allow. That's not great. 

On the other hand the whole point of these warnings is to let users know where in their dep graph are there deoptimizations. So it's not any help to not show the transitives.

To make informed decisions the user will also need the full dependency path to the commonjs dependency.

Which then leads to your specific question: should we just show deps up to a commonjs dep, then ignore that subgraph? Yeah that seems a good way to cut down on the total shown.